openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -222,7 +237,7 @@ def get_cohort(user, course_key, assign=True):
 
     user.course_groups.add(cohort)
 
-    return cohort
+    return request_cache.data.setdefault(cache_key, cohort)
```

Comment:
A `CourseUserGroup` object takes 64B according to `sys.getsizeof()`. So for a million users the caching would take 64MB. After consultation with @ormsbee I'll do a few grading runs on stage to make sure actual memory usage is inline with this expectation.